### PR TITLE
[Refactor] Express MAGI-2 mHC pre application as a batch matmul

### PR DESCRIPTION
**Pull request by yeahdongcn**
## Summary

Express MAGI-2's ordinary multi-stream mHC pre-application as a batch matrix multiplication instead of parsing an einsum expression on each call.

The coefficient expression, coefficient dtype conversion and stream contraction are unchanged. Exact-shaped multi-stream inputs use `bmm([T,1,N], [T,N,C]).squeeze(1)`; broadcast-shaped and single-stream inputs retain the original einsum path.

This avoids the historical reference #7156's elementwise multiply-then-sum rewrite, which can round low-precision products before the reduction and ignore the requested coefficient dtype. The new test includes a BF16 fixture that distinguishes that rewrite from the original matrix contraction.

## Scope and contracts

- Two files only: `MHCHandler.apply_pre` and focused tests.
- Keep sigmoid/alpha/bias, `out_dtype` coefficient-cast behavior, input validation, broadcasting and autograd behavior. As before, differing operand dtypes may error outside autocast; `out_dtype` is not a promise to override autocast's result dtype.
- No projection/normalization precision policy, parameter caching, post/mix kernel, environment variable, registry or model wiring change.
- Use an existing compiler-visible Torch operation; no CustomOp/IR/kernel registration is needed for this expression refactor.

A CPU dispatch test confirms the original einsum and explicit form each call `aten.bmm` once, with identical operand shapes, dtypes and strides. This does **not** claim fewer GPU kernels, fusion, peak-memory savings or a measured latency improvement.

## Validation

Commit `44de15740ed6709a0f3c9c61f05b286dbd9f5c97`, one signed-off commit based on upstream `a31690041de9a19acfec8239c4bbb64bc3e051d4`.

- Targeted pre-commit passed, including Ruff, mypy and test-marker checks.
- **39 CPU tests passed**, 4 MUSA deselected: exact reference comparison, FP32/FP16/BF16/FP64, 1/3/4 streams, empty/strided inputs, coefficient broadcasting and mismatched-dtype/shape errors, nonfinite values/signed zeros, gradients, and changed coefficients across Dynamo graphs.
- A complete two-layer native MAGI-2 CPU test observes four pre-application calls and matches the original output with `rtol=atol=0`.
- The Dynamo backend executes FX eagerly and asserts actual graph creation; this is not Inductor or capture validation.
- **4 MUSA tests passed** on one 48-SM MTT S5000, driver `5.2.0-server`, with **rtol=atol=0**. FP32/BF16/FP16 tests use two seeds, widths including 6144, 1/3/4 streams, zero tokens and strided streams/projection logits. The autocast test asserts actual MUSA autocast is enabled and the resulting FP16/BF16 dtype, while comparing original and explicit contractions exactly.

Stack: torch/torch_musa `2.11.0.post1+musa5.2.0`, torchada `0.1.83`, vLLM `0.28.0`, vllm-musa `0.1.28`. Exact source installed editable without dependency changes; MUSA entrypoint imports torchada first. No driver pin.

```bash
python -m pytest -q -o addopts= -m cpu \
  tests/diffusion/models/magi2/test_mhc_pre.py \
  tests/diffusion/models/magi2/test_native_preview.py
```

For MUSA, run the new test file with `-m musa` from a torchada-first entrypoint with one visible GPU.

**Not run:** CUDA hardware, GPU backward, Inductor/graph capture, full MUSA checkpoint/video or performance benchmarks. This remains Draft; it does not resolve unrelated FA3 numerical gates or qualify reduced-precision mHC projection changes.

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
This PR touches **tests/diffusion/, vllm_omni/diffusion/** (2 files). Based on CODEOWNERS coverage of the changed files, the most-related reviewers appear to be:

> @fhfuih @Bounty-hunter @wtomin

Could one of you take a look when you get a chance? Thanks!
